lib/demon/sidekiq.rb

```diff
@@ -35,9 +35,18 @@ def after_fork
 
     # Unicorn uses USR1 to indicate that log files have been rotated
     Signal.trap("USR1") do
-      log_in_trap("Sidekiq reopening logs...")
-      Unicorn::Util.reopen_logs
-      log_in_trap("Sidekiq done reopening logs...")
+      begin
+        log_in_trap("Sidekiq reopening logs...")
+        Unicorn::Util.reopen_logs
+        log_in_trap("Sidekiq done reopening logs...")
+      rescue => error
+        log_in_trap(
+          "Error encountered while reopening logs: [#{error.class}] #{error.message}\n#{error.backtrace.join("\n")}",
+          level: :error,
+        )
+
+        exit 1
```

Comment:
If any error occurs while reopening the logs, we should just exit the process so that a new process will be started.